File: packages/python/palpluss/types/wallets.py
from __future__ import annotations

from typing_extensions import TypedDict


class ServiceWalletBalance(TypedDict):
    walletId: str
    tenantId: str
    currency: str
    availableBalance: float
    ledgerBalance: float
    updatedAt: str


class ServiceTopupResponse(TypedDict):
    transactionId: str
    tenantId: str
    type: str
    status: str
    amount: float
    currency: str
    phone: str
    accountReference: str
    transactionDesc: str
    providerRequestId: str
    providerCheckoutId: str
    resultCode: str
    resultDescription: str
    idempotencyKey: str | None
    createdAt: str
    updatedAt: str

File: src/types/wallets.ts
export interface ServiceWalletBalance {
  walletId: string;
  tenantId: string;
  currency: string;
  availableBalance: number;
  ledgerBalance: number;
  updatedAt: string;
}

export interface ServiceTopupRequest {
  amount: number;
  phone: string;
  accountReference?: string;
  transactionDesc?: string;
}

export interface ServiceTopupResponse {
  transactionId: string;
  tenantId: string;
  type: 'WALLET_TOPUP_SERVICE_TOKENS';
  status: 'PENDING';
  amount: number;
  currency: string;
  phone: string;
  accountReference: string;
  transactionDesc: string;
  providerRequestId: string;
  providerCheckoutId: string;
  resultCode: string;
  resultDescription: string;
  idempotencyKey: string | null;
  createdAt: string;
  updatedAt: string;
}
